block-producer/src/server/api.rs

```diff
@@ -32,15 +34,32 @@ impl<T> api_server::Api for BlockProducerApi<T>
 where
     T: TransactionQueue,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(target = "miden-block-producer", skip(self, request), err)]
     async fn submit_proven_transaction(
         &self,
         request: tonic::Request<SubmitProvenTransactionRequest>,
     ) -> Result<tonic::Response<SubmitProvenTransactionResponse>, Status> {
         let request = request.into_inner();
+        debug!(target: target!(), tx = ?request.transaction);
 
         let tx = ProvenTransaction::read_from_bytes(&request.transaction)
             .map_err(|_| Status::invalid_argument("Invalid transaction"))?;
 
+        info!(
+            target: target!(),
+            tx_id = %tx.id().inner(),
+            account_id = %tx.account_id(),
+            initial_account_hash = %tx.initial_account_hash(),
+            final_account_hash = %tx.final_account_hash(),
+            input_notes = %format_input_notes(tx.input_notes()),
+            output_notes = %format_output_notes(tx.output_notes()),
+            tx_script_root = %format_opt(tx.tx_script_root().as_ref()),
+            block_ref = %tx.block_ref(),
+            "Deserialized transaction"
+        );
```

Comment:
Similar comment to the above. Let's serialize this as a single message rather than a set of fields.